proposals/0006-Keyboard-Friendly-Virtualization.md

```diff
@@ -0,0 +1,190 @@
+new items in batches. Batches may not render the full extent of items to
+realize, and the number of items to render per-batch is configurable.
+
+Cells in realization windows should be automatically rendered, even if not yet
+visible. This can be done during batch renders, giving priority to first render
+the window around visible content.
+
+### Endpoint-specific default behaviors
+
+VirtualizedList should be keyboard-friendly by default on desktop platforms as
+they are oriented towards usage of a physical keyboard. This additional
+realization is not free however, increasing memory usage to retain additional
+cells. This tradeoff may not make sense for mobile.
+
+This leads to desired behavior of:
+- **Keep FlatList focus-safe on desktop:** Realization windows needed for
+correct keyboard navigation should be enabled by default.
+- **Keep FlatList memory usage steady on mobile:** New realization windows
+are opt-in where not pay to play.
+
+The full matrix of behaviors is described in the *API Reference* below.
+
+Endpoint detection is generally performed via `Platform` APIs, such as
+`Platform.isTV()` or `Platform.isPad()`. This change would add a
+`Platform.isDesktop()` endpoint query to allow desktop-specific behavior.
+
+### New VirtualizedList props
+
+Properties should be added to VirtualizedList (and FlatList/SectionList) to
+allow optimizing for app-specific needs. This can be exposed as a
+`RealizationWindowConfig`, describing the size and enablement of predefined
+realization windows. This API could be expanded in the future to allow
+dynamically controlled realization behaviors.
+
+
+### API reference
+
+```tsx
+/**
+ * Allows enabling or disabling an area for realization, optionally overriding
+ * its window size.
+ */
+export type RealizationWindow =
+  | boolean
+  | [boolean, {windowSize?: number}];
+
+/**
+ * Determines what areas of the list to keep realized. Realizing extra windows
+ * may be required for correct behavior when focus is out of viewport, but may
+ * also increase memory usage.
+ */
+export type RealizationWindowConfig = {
+  /**
+   * Keeps the area around the currently focused item realized.
+   * - Enabled by default on all platforms
+   *
+   * To function with a non-default (CellRendererComponent)[
+   * https://reactnative.dev/docs/virtualizedlist#cellrenderercomponent],
+   * the `onFocusCapture` prop must be supported.
+   *
+   * Defaults to keeping the screen centered around the focused cell, along
+   * with a screen above and below realized.
+   */
+  focused?: RealizationWindow;
+
+  /**
+   * Keeps the beginning area of the list realized. Useful to allow instantly
+   * shifting focus to the first item (e.g. via Home key).
+   * - Disabled by default on all platforms (Should be enabled by default on
+   * desktop if FlatList begins transparently supporting home/end focus)
+   *
+   * Defaults to keeping a single screen at the beginning of the list realized
+   */
+  home?: RealizationWindow;
```

Comment:
I think we should call this `start` instead. The "home" terminology doesn't really fit well with our other APIs, and "start" is clear.